Sources/AWSLambdaRuntimeCore/NewLambda+JSON.swift

```diff
@@ -0,0 +1,151 @@
+    ///   - handler: The handler object.
+    package init(encoder: Encoder, decoder: Decoder, handler: Handler) where Output: Encodable {
+        self.encoder = encoder
+        self.decoder = decoder
+        self.handler = handler
+    }
+
+    /// Initializes an instance given a decoder, and a handler with a `Void` output.
+    ///   - Parameters:
+    ///   - decoder: The decoder object that will be used to decode the received ``ByteBuffer`` event into the generic ``Event`` type served to the `handler`.
+    ///   - handler: The handler object.
+    package init(decoder: Decoder, handler: Handler) where Output == Void, Encoder == VoidEncoder {
+        self.encoder = VoidEncoder()
+        self.decoder = decoder
+        self.handler = handler
+    }
+
+    /// A ``StreamingLambdaHandler/handle(_:responseWriter:context:)`` wrapper.
+    ///   - Parameters:
+    ///   - event: The received event.
+    ///   - outputWriter: The writer to write the computed response to.
+    ///   - context: The ``NewLambdaContext`` containing the invocation's metadata.
+    package mutating func handle(
+        _ request: ByteBuffer,
+        responseWriter: some LambdaResponseStreamWriter,
+        context: NewLambdaContext
+    ) async throws {
+        let event = try self.decoder.decode(Event.self, from: request)
+
+        let writer = ResponseWriter<Output>(encoder: self.encoder, streamWriter: responseWriter)
+        try await self.handler.handle(event, outputWriter: writer, context: context)
+    }
+}
+
+/// A ``LambdaResponseStreamWriter`` wrapper that conforms to ``LambdaResponseWriter``.
+package struct ResponseWriter<Output>: LambdaResponseWriter {
+    let underlyingStreamWriter: LambdaResponseStreamWriter
+    let encoder: LambdaOutputEncoder
+    var byteBuffer = ByteBuffer()
+
+    /// Initializes an instance given an encoder and an underlying ``LambdaResponseStreamWriter``.
+    /// - Parameters:
+    ///   - encoder: The encoder object that will be used to encode the generic ``Output`` into a ``ByteBuffer``, which will then be passed to `streamWriter`.
+    ///   - streamWriter: The underlying ``LambdaResponseStreamWriter`` that will be wrapped.
+    package init(encoder: LambdaOutputEncoder, streamWriter: LambdaResponseStreamWriter) {
+        self.encoder = encoder
+        self.underlyingStreamWriter = streamWriter
+    }
+
+    ///  Passes the `response` argument to ``LambdaResponseStreamWriter/writeAndFinish(_:)``.
+    /// - Parameter response: The generic ``Output`` object that will be passed to ``LambdaResponseStreamWriter/writeAndFinish(_:)``.
+    package mutating func write(response: Output) async throws {
+        if Output.self == Void.self {
+            try await self.underlyingStreamWriter.finish()
+        } else if let response = response as? Encodable {
+            try self.encoder.encode(response, into: &self.byteBuffer)
+            try await self.underlyingStreamWriter.writeAndFinish(self.byteBuffer)
+        }
```

Comment:
remove the if/else dance here. this should work thanks to removing the precondition in the voidencoder.